src/textures/Texture.js

```diff
@@ -17,77 +17,75 @@ import { ImageUtils } from '../extras/ImageUtils.js';
 
 let textureId = 0;
 
-function Texture( image, mapping, wrapS, wrapT, magFilter, minFilter, format, type, anisotropy, encoding ) {
+class Texture extends EventDispatcher {
 
-	Object.defineProperty( this, 'id', { value: textureId ++ } );
+	constructor( image, mapping, wrapS, wrapT, magFilter, minFilter, format, type, anisotropy, encoding ) {
 
-	this.uuid = MathUtils.generateUUID();
+		super();
+		// Object.defineProperty( this, 'id', { value: textureId ++ } );
+		this.id = textureId ++;
```

Comment:
`Object.defineProperty( this, 'id', { value: textureId ++ } );` was there to make that variable unmodifiable.